scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +3582,289 @@ def __rtruediv__(self, other):
+            message = "`weights` must sum to 1.0."
+            raise ValueError(message)
+
+        if not np.all(weights >= 0):
+            message = "All `weights` must be non-negative."
+            raise ValueError(message)
+
+    def __init__(self, components, /, *, weights=None):
+        self._input_validation(components, weights)
+        n = len(components)
+        dtype = np.result_type(*(var._dtype for var in components))
+        self._shape = np.broadcast_shapes(*(var._shape for var in components))
+        self._dtype, self._components = dtype, components
+        self._weights = np.full(n, 1/n, dtype=dtype) if weights is None else weights
+        self.validation_policy = None  # needed for
+
+    def _full(self, val, *args):
+        args = [np.asarray(arg) for arg in args]
+        dtype = np.result_type(self._dtype, *(arg.dtype for arg in args))
+        shape = np.broadcast_shapes(self._shape, *(arg.shape for arg in args))
+        return np.full(shape, val, dtype=dtype)
+
+    def _sum(self, fun, *args):
+        out = self._full(0, *args)
+        for var, weight in zip(self._components, self._weights):
+            out += getattr(var, fun)(*args) * weight
+        return out
+
+    def _logsum(self, fun, *args):
+        out = self._full(-np.inf, *args)
+        for var, log_weight in zip(self._components, np.log(self._weights)):
+            np.logaddexp(out, getattr(var, fun)(*args) + log_weight, out=out)
+        return out
+
+    def support(self):
+        a = self._full(np.inf)
+        b = self._full(-np.inf)
+        for var in self._components:
+            a = np.minimum(a, var.support()[0])
+            b = np.maximum(b, var.support()[1])
+        return a, b
+
+    def _raise_if_method(self, method):
+        if method is not None:
+            raise NotImplementedError("`method` not implemented for this distribution.")
+
+    def logentropy(self, *, method=None):
+        self._raise_if_method(method)
+        def log_integrand(x):
+            # `x` passed by `_tanhsinh` will be of complex dtype because
+            # `log_integrand` returns complex values, but the imaginary
+            # component is always zero. Extract the real part because
+            # `logpdf` uses `logaddexp`, which fails for complex input.
+            return self.logpdf(x.real) + np.log(self.logpdf(x.real) + 0j)
+
+        res = _tanhsinh(log_integrand, *self.support(), log=True).integral
+        return _log_real_standardize(res + np.pi*1j)
+
+    def entropy(self, *, method=None):
+        self._raise_if_method(method)
+        return _tanhsinh(lambda x: -self.pdf(x) * self.logpdf(x),
+                         *self.support()).integral
+
+    def mode(self, *, method=None):
+        self._raise_if_method(method)
+        a, b = self.support()
+        f = lambda x: -self.pdf(x)  # noqa: E731 is silly
+        res = _bracket_minimum(f, 1., xmin=a, xmax=b)
+        res = _chandrupatla_minimize(f, res.xl, res.xm, res.xr)
```

Comment:
This is a case where it would be cool to be able to find a global minimum here and find the actual mode, if there is one. There are some heuristics for the initial guess we could use if we assume the components are all unimodal, but they wouldn't be fool proof. One I have in mind is finding the finite set of calculated modes of all of the components, and then finding the maximum of the mixture pdf among these, and using it as the initial guess. I'm still unsure if it would be better to implement an imperfect heuristic which would plausibly get the right result in most cases, or do something like give the user freedom to pick the initial guess. I think it's fine to punt on this, but doing something about making mode work better for mixtures of unimodal distributions is on my nice-to-have list.

Reply:
I think the approach we've discussed to find all roots of a function would be useful here: adaptively fit a spline to the pdf, then find all roots of the spine's derivative. 

This is one of the reasons I want our splines to support N-D arrays for the abscissa.